### PR TITLE
[github-actions] add GitHub Actions

### PR DESCRIPTION
**Pull request by wgtdkp**
This PR adds GitHub Actions.

- [X] Github Action yaml files.
- [X]  rebase on PR https://github.com/openthread/ot-commissioner/pull/87 to fix Android build failure.

### CODE REVIEW

**Comment by jwhui** [comment]
@wgtdkp , does this replace the need for Travis?

**Comment by wgtdkp** [comment]
> @wgtdkp , does this replace the need for Travis?

@jwhui Yes, what about removing travis after merging github action?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -52,6 +52,8 @@ jobs:
           sudo apt --no-install-recommends install -y clang-tools clang-format-6.0
           clang-format-6.0 --version
           python3 -m pip install yapf==0.29.0
+          suto apt-get install -y npm
+          sudo npm i -g npx
```
Comment (jwhui) [inline .github/workflows/build.yml:56]: Just curious - was this necessary for GitHub Actions?

Comment (wgtdkp) [reply .github/workflows/build.yml]: It seems GitHub Action already installed `npx`, reverted.

**Pull request merged**